prompting/rewards/date.py

```diff
@@ -20,50 +20,47 @@ def date_diff(self, ref_date: tuple, comp_date: tuple) -> int:
         """
         Calculates the absolute difference in days between two dates.
         """
+        if not comp_date:
+            return 9999
+        # Check if ref date is just a year
+        if ref_date.isdigit():
+            # Extract the last 3-4 digits from the completion date using a regex pattern that would detect 3 or 4 digit years 
+            comp_year = re.findall(r'\b\d{3,4}\b', comp_date)
+            if comp_year:
+                return abs(int(ref_date) - int(comp_year[0])*365)
+            else:
+                return 9999
```

Comment:
no need to refactor but it would be nice to have a global variable set to 9999 for clarity purposes